### PR TITLE
CosmosDB Plugin

### PR DESCRIPTION
**Pull request by deadlyfingers**

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -291,5 +291,9 @@ router.post('/setup', (req, res) => {
 });
 
 module.exports = {
-  router
+  router,
+  paths,
```
Comment (morsh) [inline server/routes/api.js]: Can you please put all of these under a separate file of common server utilities?

File: server/routes/cosmos-db.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+const router = new express.Router();
+const crypto = require('crypto');
+
+const paths = require('./api').paths, 
+      getFileContents = require('./api').getFileContents,
+      isValidFile = require('./api').isValidFile, 
+      getField = require('./api').getField;
+
+router.post('/dashboard/:id', (req, res) => {
+  const { id } = req.params;
```
Comment (morsh) [inline server/routes/cosmos-db.js]: why use the dashboard api path and not create a new one?

File: src/data-sources/plugins/CosmosDB/Query.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    const paths = location.pathname.split('/');
+    if (paths.length !== 3) {
+      throw Error('Expected location pathname:' + paths);
+    }
+    const dashboardId = paths[paths.length - 1];
+
+    const params = this._props.params;
+    const query: string = this.compileQuery(params.query, dependencies);
+
+    const url = `/cosmosdb/dashboard/${dashboardId}`;
```
Comment (morsh) [inline src/data-sources/plugins/CosmosDB/Query.ts]: I think you can change this to be a url that receives all of its parameters in the post parameters (no need for the server to poll the parameters from the dashboard itself)

We can discuss this to see what makes sense.

**Comment by deadlyfingers** [comment]
- Added readme md
- Moved connection logic from server API to client Query and therefore removed need for any shared util methods.

**Pull request merged**